### PR TITLE
add simple CI workflow

### PR DESCRIPTION
**Pull request by evgeni**
____________________________________________________________________

Your contributions to Redmine are welcome!

Please **open an issue on the [official website]** instead of sending pull requests.

Since the development of Redmine is not conducted on GitHub but on the [official website] and core developers are not monitoring the GitHub repo, pull requests might not get reviewed.

For more detail about how to contribute, please see the wiki page [Contribute] on the [official website].

[official website]: https://www.redmine.org/
[Contribute]: https://www.redmine.org/projects/redmine/wiki/Contribute

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Remove Gemfile.lock
+        run: rm -f Gemfile.lock
+      - name: Setup Database
+        run: "printf 'test:\n  adapter: sqlite3\n  database: db/redmine.sqlite3' > config/database.yml"
+      - name: Setup secure config
+        run: printf '{"data_dir":"/var/lib/redmine"}' | sudo tee /etc/redmine/secure_config.yaml
```
Comment (ekohl) [inline .github/workflows/ci.yml:35]: Needs a `mkdir -p /etc/redmine`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+jobs:
+  ruby:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby:
+          - '2.7'
+          - '3.0'
+          - '3.1'
```
Comment (ekohl) [inline .github/workflows/ci.yml:25]: Do we care about versions other than what we run?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Remove Gemfile.lock
+        run: rm -f Gemfile.lock
```
Comment (ekohl) [inline .github/workflows/ci.yml:31]: Wouldn't it make sense to run with the lock file? I'm guessing you want to add dependabot for gems too.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (evgeni) [reply .github/workflows/ci.yml:26]: future? probably
past? probably not

**Comment by evgeni** [comment]
if only this would pass…

**Comment by evgeni** [comment]
this is funny, the tests are actually broken in 5.0.5: https://github.com/theforeman/redmine/actions/runs/6455647725
but fixed in 5.0.6 (which is the version I want to rebase ontop): https://github.com/theforeman/redmine/actions/runs/6455654475

**Comment by evgeni** [comment]
I intend to fix the tests with the merge of 5.0.6 later

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      matrix:
+        ruby:
+          - '3.0'
+          - '3.1'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Install dependencies
+        run: sudo apt-get install -y gsfonts
```
Comment (ekohl) [inline .github/workflows/ci.yml:33]: It surprises me you don't need to run update too.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Archive Gemfile.lock
+        uses: actions/upload-artifact@v3
+        with:
+          name: Gemfile.lock
+          path: Gemfile.lock
```
Comment (ekohl) [inline .github/workflows/ci.yml:49]: It's stored in git, so is this this needed now that you don't run `rm -f` anymore?

File: .github/workflows/ci.yml (hunk 4 above)
Comment (evgeni) [reply .github/workflows/ci.yml:33]: it runs on the GHA "thing", which does have a populated cache (vs a "real" container that has no apt cache)

**Pull request merged**